src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIQueueWebsocketService.java

```diff
@@ -113,19 +125,22 @@ private class BuildAgentListener implements EntryAddedListener<String, LocalCIBu
         @Override
         public void entryAdded(com.hazelcast.core.EntryEvent<String, LocalCIBuildAgentInformation> event) {
             log.debug("Build agent added: {}", event.getValue());
-            sendBuildAgentInformationOverWebsocket();
+            sendBuildAgentSummaryOverWebsocket();
+            sendBuildAgentDetailsOverWebsocket(event.getValue().name());
```

Comment:
Since this is repeated 3 times I would propose that `sendBuildAgentSummaryOverWebsocket` and `sendBuildAgentDetailsOverWebsocket` are wrapped in the method `sendBuildAgentInformationOverWebsocket` again